controllers/util/status.go

```diff
@@ -239,6 +241,18 @@ func setStatusConditionIfNotPresent(conditions *[]conditionsv1.Condition, condit
 	conditionsv1.SetStatusCondition(conditions, condition)
 }
 
+// update an existing condition only if its already present and the status
+// is different or the condition is not present and we want to add a new status
+func setStatusConditionIfDifferentOrNotPresent(conditions *[]conditionsv1.Condition, condition conditionsv1.Condition) {
```

Comment:
IMO the condition should be always present Either the state is good or bad. Someone might be looking for the condition specifically. most of the status automation is done based on the conditions themselves. So I think we can just keep the `SetVersionMismatchCondition` func and it will always set the condition without any if condition.

Reply:
Updated the code accordingly, PTAL.